packages/amplify-provider-awscloudformation/src/aws-utils/aws-s3.ts

```diff
@@ -126,6 +128,14 @@ export class S3 {
         spinner.start('Uploading files.');
       }
       logger('uploadFile.s3.upload', [others])();
+      const minChunkSize = 5 * 1024 * 1024; // 5 MB
+      if (augmentedS3Params.Body instanceof fs.ReadStream && fs.statSync(augmentedS3Params.Body.path).size <= minChunkSize) {
+        // Buffer small files to avoid memory leak.
```

Comment:
Why buffering small files help? Shouldn't we buffer big files?

Reply:
This is because before this change https://github.com/aws-amplify/amplify-cli/pull/13493 all files that were less than 5mb were uploaded via `s3.putObject` API. 
I.e. population of files smaller than 5mb was NOT running into memory leak.
The memory leak only happens if `Body` is a stream, so buffering small files brings back former balance between uploads that leak and not leak.
(keep in mind that large files always leaked, but there wasn't a lot of them, at least not enough to cause current problem we have now).